components/base/sensorcontrolled/velocities.go

```diff
@@ -0,0 +1,135 @@
+package sensorcontrolled
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/control"
+	"go.viam.com/rdk/utils"
+)
+
+type linearVelControllable struct {
+	control.Controllable // is this necessary?
+	sb                   base.Base
+	ms                   movementsensor.MovementSensor
+}
+
+func (l *linearVelControllable) SetState(ctx context.Context, state float64) error {
```

Comment:
[q] These are two separate loops, I may need to combine the set state function or have a helper function feed into the unexported function called by SetVelocity, so not to zero-out a linear call when an angular is commanded.